node/src/components/storage.rs

```diff
@@ -1517,10 +1517,15 @@ fn initialize_block_body_merkle_db(
     deploy_hashes_db: &Database,
     transfer_hashes_db: &Database,
     proposer_db: &Database,
-    deleted_block_hashes: &HashSet<&[u8]>,
+    deleted_block_body_hashes: &HashSet<&[u8]>,
     should_check_integrity: bool,
 ) -> Result<(), LmdbExtError> {
     info!("initializing block body (Merkle) database");
+    if deleted_block_body_hashes.is_empty() {
+        info!("No deleted block bodies, nothing to do.");
+        return Ok(());
```

Comment:
Then we'll also skip the integrity check, though.

Reply:
Just FYI, we do employ that approach in other storage(s) already: https://github.com/casper-network/casper-node/pull/1508

Reply:
A partial fix was introduced in https://github.com/casper-network/casper-node/pull/1563/commits/c00026876a204c10ff5b8fc492352218287d8e86

Reply:
Additional integrity checking is necessary.  I have ticked it here: https://github.com/casper-network/casper-node/issues/1610